lib/cylc/config.py

```diff
@@ -1085,12 +1085,14 @@ def configure_queues(self):
         # Then reassign to other queues as requested.
         warnings = []
         requeued = []
-        for queue in queues:
-            if queue == 'default':
+        for key, queue in queues.copy().items():
+            # queues.copy() is essential here to allow items to be removed from
+            # the queues dict.
```

Comment:
I think a slightly simpler slightly more efficient alternative here would be to iterative over `dict.keys()` which returns just a list of keys (which does not change if the original dict changes inside the loop), and use that to index into the dict?